Graphics/GraphicsEngine/interface/GraphicsTypes.h

```diff
@@ -1271,6 +1272,17 @@ struct DeviceObjectAttribs
     explicit constexpr DeviceObjectAttribs(const Char* _Name) :
         Name{_Name}
     {}
+
+    /// Comparison operator tests if two structures are equivalent
+
+    /// \param [in] RHS - reference to the structure to perform comparison with
+    /// \return
+    /// - True if all members of the two structures are equal.
+    /// - False otherwise.
+    bool operator==(const DeviceObjectAttribs& RHS) const
+    {
+        return SafeStrEqual(Name, RHS.Name);
+    }
```

Comment:
Objects with different names are considered the same if other fields are the same. E.g. two rasterizer states "Rasterizer State 1" and "Rasterizer State 2" are still the same if other fields are identical.
Several internal caching systems rely on that and this may break them.